js/runtime.js

```diff
@@ -818,6 +836,56 @@
                     }
                 }, points);
             },
+            regularPolygon: function regularPolygon(origin, noSides, radius){
+                var points = [];
+                util.setLastPoint(origin);
+                var dA = 360 / noSides;
+                for (var i = 0; i < noSides; i++){
+                    var angle = util.deg2rad(i * dA);
+                    points.push(util.add(origin, new util.Vector(
+                        Math.cos(angle) * radius,
+                        Math.sin(angle) * radius)));
+                }
+                return new util.Shape(function(ctx){
+                    var start = points[noSides-1];
+                    if (!util.isDrawingPath()){
+                        ctx.beginPath();
+                        ctx.moveTo(start.x, start.y);
+                    }else{
+                        ctx.moveTo(start.x, start.y);
```

Comment:
This call to `moveTo` breaks the context for the shape. This gist shows that: b0ec4a40e14ab4a4c817. The shape should be closed but isn't. Replacing `moveTo()` with `lineTo()` fixes it